pandas/io/parsers.py

```diff
@@ -1296,15 +1296,28 @@ def _validate_usecols_arg(usecols):
     if usecols is not None:
         if callable(usecols):
             return usecols, None
-        # GH20529, ensure is iterable container but not string.
-        elif not is_list_like(usecols):
+
+        if not is_list_like(usecols):
+            # see gh-20529
+            #
+            # Ensure it is iterable container but not string.
             raise ValueError(msg)
-        else:
-            usecols_dtype = lib.infer_dtype(usecols, skipna=False)
-            if usecols_dtype not in ('empty', 'integer',
-                                     'string', 'unicode'):
-                raise ValueError(msg)
-        return set(usecols), usecols_dtype
+
+        usecols_dtype = lib.infer_dtype(usecols, skipna=False)
+
+        if usecols_dtype not in ("empty", "integer",
+                                 "string", "unicode"):
+            raise ValueError(msg)
+
+        usecols = set(usecols)
+
+        if usecols_dtype == "unicode":
+            # see gh-13253
+            #
+            # Python 2.x compatibility
+            usecols = {col.encode("utf-8") for col in usecols}
```

Comment:
Hmm shouldn't we just use whatever coding is supplied to the higher level parsing function?

Reply:
Good question, but no actually.  The `encoding` argument is used to properly load the data from file source into Python (more so in Python 3.x than in Python 2.x).

That being said, a more robust way is to prefer the `sys.getfilesytemencoding` before defaulting to `utf-8` in this situation.

Reply:
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=7144

Actually, I take that back.  It looks `utf-8` is the away to go, at least to appease CI.